tests/data/test_datadoc_extract.py

```diff
@@ -0,0 +1,507 @@
+        <meta:ContextVariable id="2">
+            <codelist:Codelist>
+                <codelist:CodelistMeta>
+                    <codelist:Title>Codelist Title 2</codelist:Title>
+                    <codelist:Description>Codelist Description 2</codelist:Description>
+                </codelist:CodelistMeta>
+                <codelist:Codes>
+                    <codelist:Code>
+                        <codelist:CodeValue>003</codelist:CodeValue>
+                        <codelist:CodeText>Code Text 3</codelist:CodeText>
+                    </codelist:Code>
+                </codelist:Codes>
+            </codelist:Codelist>
+        </meta:ContextVariable>
+    </root>
+    """
+    root = ET.fromstring(xml_data)
+    codelist_data = datadok_extract.extract_codelist(root)
+
+    assert len(codelist_data) == 3
+    assert codelist_data[0].context_id == "1"
+    assert codelist_data[0].codelist_title == "Codelist Title 1"
+    assert codelist_data[0].codelist_description == "Codelist Description 1"
+    assert codelist_data[0].code_value == "001"
+    assert codelist_data[0].code_text == "Code Text 1"
+
+    assert codelist_data[2].context_id == "2"
+    assert codelist_data[2].codelist_title == "Codelist Title 2"
+    assert codelist_data[2].codelist_description == "Codelist Description 2"
+    assert codelist_data[2].code_value == "003"
+    assert codelist_data[2].code_text == "Code Text 3"
+
+
+def test_extract_codelist_missing_elements():
+    xml_data_missing = """
+    <root xmlns:meta="http://www.ssb.no/ns/meta" xmlns:codelist="http://www.ssb.no/ns/meta/codelist">
+        <meta:ContextVariable id="1">
+            <codelist:Codelist>
+                <codelist:CodelistMeta>
+                    <!-- Missing title and description -->
+                </codelist:CodelistMeta>
+                <codelist:Codes>
+                    <codelist:Code>
+                        <!-- Missing code value and code text -->
+                    </codelist:Code>
+                </codelist:Codes>
+            </codelist:Codelist>
+        </meta:ContextVariable>
+    </root>
+    """
+    root = ET.fromstring(xml_data_missing)
+    codelist_data = datadok_extract.extract_codelist(root)
+
+    # Since the codelist is missing critical elements, the result should be empty
+    assert len(codelist_data) == 0
+
+
+def test_handle_decimals_with_comma():
+    # Test case with comma as the decimal separator
+    data = {"col1": ["1,23", "4,56", "7,89"], "col2": ["10,00", "20,50", "30,75"]}
+    df = pd.DataFrame(data)
+
+    metadata_data = {
+        "title": ["col1", "col2"],
+        "datatype": ["Desimaltall", "Desimaltall"],
+        "precision": [None, None],  # Precision not used in this case
+    }
+    metadata_df = pd.DataFrame(metadata_data)
+
+    df, metadata_df = datadok_extract.handle_decimals(df, metadata_df)
+
+    assert df["col1"].dtype == "Float64"
+    assert df["col2"].dtype == "Float64"
+    assert df["col1"].iloc[0] == 1.23
```

Comment:
Equality checks with floating points values are unsafe, see https://rules.sonarsource.com/python/RSPEC-1244/
Use `math.isclose` to solve this, like:
```python
assert math.isclose(df["col1"].iloc[0], 1.23)
```

Reply:
Nice thanks, will do